components/post_view/channel_intro_message/add_members_button.tsx

```diff
@@ -37,9 +41,21 @@ const AddMembersButton: React.FC<AddMembersButtonProps> = ({totalUsers, usersLim
 
     const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;
     const inviteUsers = totalUsers < usersLimit;
+    const currentTeamId = useSelector(getCurrentTeamId);
+    const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
+
+
+    if (isGuest(currentUser)) {
+        return null;
+    }
 
     return (
-        inviteUsers && !isPrivate ? lessThanMaxFreeUsers(setHeader) : moreThanMaxFreeUsers(channel, setHeader)
+        <TeamPermissionGate
+            teamId={currentTeamId}
+            permissions={[Permissions.ADD_USER_TO_TEAM, Permissions.INVITE_GUEST]}
+        >
+            {inviteUsers && !isPrivate ? lessThanMaxFreeUsers(setHeader) : moreThanMaxFreeUsers(channel, setHeader)}
+        </TeamPermissionGate>
```

Comment:
We can test this, but I believe he `TeamPearmissonGate` will take care of the guest users. Guests probably don't have the `Permissions.ADD_USER_TO_TEAM, Permissions.INVITE_GUEST` permissions so we won't need to check if `isGuest` first

Reply:
ok, that makes sense.